airflow/ui/src/pages/DagsList/DagsFilters.tsx

```diff
@@ -25,11 +25,17 @@ import { useTableURLState } from "src/components/DataTable/useTableUrlState";
 import { QuickFilterButton } from "src/components/QuickFilterButton";
 
 const PAUSED_PARAM = "paused";
+const STATE_PARAM = "lastrun";
 
 export const DagsFilters = () => {
   const [searchParams, setSearchParams] = useSearchParams();
 
   const showPaused = searchParams.get(PAUSED_PARAM);
+  const state = searchParams.get(STATE_PARAM);
+  const isAll = state === null || state === "all";
+  const isRunning = Boolean(state) && state === "running";
+  const isFailed = Boolean(state) && state === "failed";
+  const isSuccess = Boolean(state) && state === "success";
```

Comment:
null & undefined are !== anyway
```suggestion
  const isRunning = state === "running";
  const isFailed = state === "failed";
  const isSuccess = state === "success";
```